src/main/kotlin/org/rust/cargo/toolchain/WslRustToolchain.kt

```diff
@@ -0,0 +1,79 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.cargo.toolchain
+
+import com.intellij.execution.configurations.GeneralCommandLine
+import org.intellij.lang.annotations.Language
+import org.rust.openapiext.withWorkDirectory
+import java.nio.file.Files
+import java.nio.file.Path
+
+class WslRustToolchain(location: Path) : RustToolchain(location) {
+    private val distribution: String
+    private val basePath: String
+
+    init {
+        val groups = toolchainPathRegex.find(location.toString())!!.groupValues
+        distribution = groups[1]
+        basePath = "/${groups[2].replace("\\", "/")}"
+    }
+
+    override fun getSysroot(projectDirectory: Path): String? {
+        val original = super.getSysroot(projectDirectory)
+        return executeBashCommand("wslpath -w $original")
+    }
+
+    override fun createGeneralCommandLine(tool: String, vararg arguments: String, workingDirectory: Path?, setup: GeneralCommandLine.() -> Unit): GeneralCommandLine {
+        val transform: (String) -> String = {
+            if ("^[a-z]:\\\\(?:[^\\\\/:*?\"<>|\\r\\n]+\\\\)*[^\\\\/:*?\"<>|\\r\\n]*\$".toRegex().matches(it))
+                windowsToLinuxPath(it.replace("\\", "\\\\"))
+            else it
+        }
+        val generalCommandLine = GeneralCommandLine("wsl.exe")
+            .withCharset(Charsets.UTF_8)
+            .withWorkDirectory(workingDirectory)
+            .withParameters("$basePath/$tool", *arguments.map(transform).toTypedArray())
+        generalCommandLine.setup()
+        return generalCommandLine
+    }
+
+    override fun hasExecutable(name: String) = Files.exists(location.resolve(name))
+
+    override val isRustupAvailable: Boolean
+        get() = hasExecutable(RUSTUP)
+
+    override val presentableLocation: String
+        get() = location.resolve(CARGO).toString()
+
+    override fun processOutput(original: String): String {
+        return transformWslPaths(original)
+    }
+
+    companion object {
+        @Language("RegExp")
+        val toolchainPathRegex = """^\\\\wsl\$\\(\w+)\\(.+)$""".toRegex()
+
+        fun executeBashCommand(command: String): String {
+            val process = Runtime.getRuntime().exec("""wsl $command""")
```

Comment:
```suggestion
            val process = Runtime.getRuntime().exec("""wsl -d $distribution $command""")
```
This shouldn't be a companion object because it needs to access the `distribution` property of WslRustToolchain.